docs/site/Deployment-guide-Cloud-Foundy.md

```diff
@@ -0,0 +1,244 @@
+binding.
+
+```ts
+import {TodoListApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+const datasourceDb = require('./datasources/db.datasource.json');
+
+type VcapService = {
+  name: string;
+  credentials: {
+    // tslint:disable-next-line:no-any
+    [property: string]: any;
+  };
+};
+
+if (shouldConfigureForCf()) {
+  const services = JSON.parse(process.env.VCAP_SERVICES!);
+  Object.keys(services).forEach(key => {
+    const serviceGroup = services[key];
+
+    // tslint:disable-next-line:no-any
+    serviceGroup.forEach((service: VcapService) => {
+      if (service.name === datasourceDb.vcapServiceName) {
+        datasourceDb.url = service.credentials.url;
+      }
+    });
+  });
+}
+
+export async function main(options?: ApplicationConfig) {
+  // IBM Cloud app port is set on process.env.PORT
+  if (!options) {
+    options = {rest: {port: process.env.PORT}};
+  } else {
+    if (!options.rest) options.rest = {};
+    options.rest.port = process.env.PORT || options.rest.port;
+  }
+  const app = new TodoListApplication(options);
+
+  await app.boot();
+  await app.bind('datasources.config.db').to(datasourceDb);
+  await app.start();
+
+  const url = app.restServer.url;
+  console.log(`Server is running at ${url}`);
+  return app;
+}
+
+function shouldConfigureForCf() {
+  if (
+    datasourceDb.vcapServiceName && // have to explicitly config datasource to mark it as an IBM Cloud service with a matching name locally
+    process.env.VCAP_SERVICES // service should have been provisioned and bound to the app
+  ) {
+    return true;
+  }
+  return false;
+}
+```
+
+## Cloud setup
+
+We will use [IBM Cloud](https://www.ibm.com/cloud/)'s Cloud Foundry for this
+example.
+
+### Database
+
+1.  Provision a Cloudant service, and name it `myCloudantDb`, which is the same
+    as the `vcapServiceName`'s value configured in `db.datasource.json`.
+2.  Log on to the Cloudant Dashboard and create a database named `todo`.
+
+### Connecting the database to app
+
+Before we can use the provisioned database, we have to connect the database to
+an app on Cloud Foundry.
+
+So, create a
+[Node.js SDK app](https://console.bluemix.net/catalog/starters/sdk-for-nodejs)
+and connect it to the provisioned Cloudant service.
+
+Update the details on `manifest.yml` with the details from the created app's
+dashboard.
+
+## Deploying the app
+
+### Local
+
+1.  Install all dependencies:
+
+```sh
+npm i
+```
+
+2.  Build the app:
+
+```sh
+npm run build
+```
+
+3.  Ensure Cloudant is running locally.
+4.  Start the app:
+
+```sh
+npm start
+```
+
+5.  Go to http://localhost:3000/swagger-ui to load API Explorer and add a todo
+    or two.
+
+6.  Go to http://localhost:3000/todos to see the todos.
+
+### Cloud
+
+1.  Build the app locally:
+
+```sh
+npm run build
+```
+
+2.  Push the app to Cloud Foundry:
+
+```
+cf push mylb4app
+```
+
+3.  Go to https://mylb4app.eu-gb.mybluemix.net/swagger-ui to load the API
```

Comment:
- might want to explain a bit that the link is derived from the name and domain in the manifest.yml file.